The `curator` client dependency in the `Zookeeper` registry document is missing

您好！文中使用 curator 客户端一节仅引入依赖curator-framework和curator-recipes。
文档定位：[Zookeeper 注册中心](https://dubbo.apache.org/zh/docs/v3.0/references/configuration/references/registry/zookeeper/)
在使用dubbo-spring-boot-starter3.0.0时（JDK为Dragonwell11.0.10+0，以spring-boot-starter-web2.5.2，zookeeper3.7.0，zkclient0.1，curator-framework4.3.0和curator-recipes4.3.0进行测试），还需要额外引入curator-x-discovery（以curator-x-discovery4.3.0作为测试），否则抛出异常：

````
Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-07-02 11:51:16.532 ERROR 15164 --- [           main] o.s.boot.SpringApplication               : Application run failed

java.lang.RuntimeException: java.lang.NoClassDefFoundError: org/apache/curator/x/discovery/ServiceDiscoveryBuilder
	at org.apache.dubbo.common.function.ThrowableAction.execute(ThrowableAction.java:48) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.client.ServiceDiscoveryRegistry.createServiceDiscovery(ServiceDiscoveryRegistry.java:129) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.client.ServiceDiscoveryRegistry.<init>(ServiceDiscoveryRegistry.java:113) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.client.ServiceDiscoveryRegistryFactory.createRegistry(ServiceDiscoveryRegistryFactory.java:35) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.support.AbstractRegistryFactory.getRegistry(AbstractRegistryFactory.java:159) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.RegistryFactoryWrapper.getRegistry(RegistryFactoryWrapper.java:34) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.RegistryFactory$Adaptive.getRegistry(RegistryFactory$Adaptive.java) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.integration.RegistryProtocol.getRegistry(RegistryProtocol.java:358) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.integration.RegistryProtocol.export(RegistryProtocol.java:213) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.rpc.protocol.ProtocolListenerWrapper.export(ProtocolListenerWrapper.java:64) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.rpc.cluster.filter.ProtocolFilterWrapper.export(ProtocolFilterWrapper.java:60) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.qos.protocol.QosProtocolWrapper.export(QosProtocolWrapper.java:64) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.rpc.Protocol$Adaptive.export(Protocol$Adaptive.java) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.config.ServiceConfig.doExportUrlsFor1Protocol(ServiceConfig.java:511) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.config.ServiceConfig.doExportUrls(ServiceConfig.java:344) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.config.ServiceConfig.doExport(ServiceConfig.java:320) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.config.ServiceConfig.export(ServiceConfig.java:219) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.exportServices(DubboBootstrap.java:1295) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.start(DubboBootstrap.java:1090) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.config.spring.context.DubboBootstrapApplicationListener.onContextRefreshedEvent(DubboBootstrapApplicationListener.java:70) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.config.spring.context.DubboBootstrapApplicationListener.onApplicationContextEvent(DubboBootstrapApplicationListener.java:63) ~[dubbo-3.0.0.jar:3.0.0]
	at com.alibaba.spring.context.OnceApplicationContextEventListener.onApplicationEvent(OnceApplicationContextEventListener.java:52) ~[spring-context-support-1.0.8.jar:na]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:176) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:169) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:143) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:421) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:378) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:938) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:586) ~[spring-context-5.3.8.jar:5.3.8]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:145) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:434) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:338) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1343) ~[spring-boot-2.5.2.jar:2.5.2]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1332) ~[spring-boot-2.5.2.jar:2.5.2]
	at com.lingh.ProviderServerApplication.main(ProviderServerApplication.java:9) ~[classes/:na]
Caused by: java.lang.NoClassDefFoundError: org/apache/curator/x/discovery/ServiceDiscoveryBuilder
	at org.apache.dubbo.registry.zookeeper.util.CuratorFrameworkUtils.buildServiceDiscovery(CuratorFrameworkUtils.java:53) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.zookeeper.ZookeeperServiceDiscovery.doInitialize(ZookeeperServiceDiscovery.java:79) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.client.AbstractServiceDiscovery.initialize(AbstractServiceDiscovery.java:38) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.registry.client.ServiceDiscoveryRegistry.lambda$createServiceDiscovery$0(ServiceDiscoveryRegistry.java:130) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.common.function.ThrowableAction.execute(ThrowableAction.java:46) ~[dubbo-3.0.0.jar:3.0.0]
	... 35 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.apache.curator.x.discovery.ServiceDiscoveryBuilder
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581) ~[na:na]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[na:na]
	... 40 common frames omitted

2021-07-02 11:51:20.800  INFO 15164 --- [           main] f.a.ReferenceAnnotationBeanPostProcessor : class org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor was destroying!

Process finished with exit code 1
```

--- a/content/zh/docs/references/registry/zookeeper.md
+++ b/content/zh/docs/references/registry/zookeeper.md
@@ -104,18 +104,23 @@ zookeeper://10.20.153.10:2181?client=curator
 ```xml
 <properties>
     <dubbo.version>2.7.8</dubbo.version>
-    <zookeeper.version>2.12.0</zookeeper.version>
+    <curator.version>5.2.1</curator.version>
 </properties>
 
 <dependency>
     <groupId>org.apache.curator</groupId>
     <artifactId>curator-framework</artifactId>
-    <version>${zookeeper.version}</version>
+    <version>${curator.version}</version>
 </dependency>
 <dependency>
     <groupId>org.apache.curator</groupId>
     <artifactId>curator-recipes</artifactId>
-    <version>${zookeeper.version}</version>
+    <version>${curator.version}</version>
+</dependency>
+<dependency>
+    <groupId>org.apache.curator</groupId>
+    <artifactId>curator-x-discovery</artifactId>
+    <version>${curator.version}</version>
 </dependency>
 ```
 

--- a/content/zh/docs/references/registry/zookeeper.md
+++ b/content/zh/docs/references/registry/zookeeper.md
@@ -34,7 +34,7 @@ description: "Zookeeper 注册中心参考手册"
 <dependency>
     <groupId>org.apache.zookeeper</groupId>
     <artifactId>zookeeper</artifactId>
-    <version>3.3.3</version>
+    <version>3.8.0</version>
 </dependency>
 ```
 
@@ -72,7 +72,7 @@ zookeeper://10.20.153.10:2181?client=zkclient
 <dependency>
     <groupId>com.github.sgroschupf</groupId>
     <artifactId>zkclient</artifactId>
-    <version>0.1</version>
+    <version>0.11</version>
 </dependency>
 ```
 
@@ -103,7 +103,6 @@ zookeeper://10.20.153.10:2181?client=curator
 
 ```xml
 <properties>
-    <dubbo.version>2.7.8</dubbo.version>
     <curator.version>5.2.1</curator.version>
 </properties>
 

--- a/content/zh/docsv2.7/user/references/registry/zookeeper.md
+++ b/content/zh/docsv2.7/user/references/registry/zookeeper.md
@@ -104,18 +104,23 @@ zookeeper://10.20.153.10:2181?client=curator
 ```xml
 <properties>
     <dubbo.version>2.7.8</dubbo.version>
-    <zookeeper.version>2.12.0</zookeeper.version>
+    <curator.version>5.2.1</curator.version>
 </properties>
 
 <dependency>
     <groupId>org.apache.curator</groupId>
     <artifactId>curator-framework</artifactId>
-    <version>${zookeeper.version}</version>
+    <version>${curator.version}</version>
 </dependency>
 <dependency>
     <groupId>org.apache.curator</groupId>
     <artifactId>curator-recipes</artifactId>
-    <version>${zookeeper.version}</version>
+    <version>${curator.version}</version>
+</dependency>
+<dependency>
+    <groupId>org.apache.curator</groupId>
+    <artifactId>curator-x-discovery</artifactId>
+    <version>${curator.version}</version>
 </dependency>
 ```
 

--- a/content/zh/docsv2.7/user/references/registry/zookeeper.md
+++ b/content/zh/docsv2.7/user/references/registry/zookeeper.md
@@ -34,7 +34,7 @@ description: "Zookeeper 注册中心参考手册"
 <dependency>
     <groupId>org.apache.zookeeper</groupId>
     <artifactId>zookeeper</artifactId>
-    <version>3.3.3</version>
+    <version>3.8.0</version>
 </dependency>
 ```
 
@@ -72,7 +72,7 @@ zookeeper://10.20.153.10:2181?client=zkclient
 <dependency>
     <groupId>com.github.sgroschupf</groupId>
     <artifactId>zkclient</artifactId>
-    <version>0.1</version>
+    <version>0.11</version>
 </dependency>
 ```
 
@@ -103,7 +103,6 @@ zookeeper://10.20.153.10:2181?client=curator
 
 ```xml
 <properties>
-    <dubbo.version>2.7.8</dubbo.version>
     <curator.version>5.2.1</curator.version>
 </properties>
 

